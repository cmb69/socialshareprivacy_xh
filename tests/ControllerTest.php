<?php

namespace Socialshareprivacy;

use ApprovalTests\Approvals;
use PHPUnit\Framework\MockObject\MockObject;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\Jquery;

class ControllerTest extends TestCase
{
    /** @var Jquery&MockObject */
    private $jquery;

    protected function setUp(): void
    {
        global $plugin_cf, $plugin_tx;
        $plugin_cf = XH_includeVar("./config/config.php", "plugin_cf");
        $plugin_tx = XH_includeVar("./languages/en.php", "plugin_tx");
        $this->jquery = $this->createMock(Jquery::class);
    }

    private function sut(): Controller
    {
        return new Controller(
            "./",
            $this->jquery
        );
    }

    public function testRendersScripts(): void
    {
        $this->jquery->expects($this->once())->method("include");
        $this->jquery->expects($this->once())->method("includePlugin")->with("socialshareprivacy");
        $request = new FakeRequest();
        $response = $this->sut()->init($request);
        Approvals::verifyHtml($response->bjs());
    }
}
