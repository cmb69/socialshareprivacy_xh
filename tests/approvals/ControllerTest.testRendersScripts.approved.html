<script type="text/javascript">/* <![CDATA[ */jQuery(function() {jQuery(".socialshareprivacy").socialSharePrivacy({"info_link":"http:\/\/www.h-online.com\/features\/Two-clicks-for-more-privacy-1783256.html","txt_help":"If you activate these buttons by clicking, information will be sent to Facebook, Twitter resp. Google in the USA and perhaps stored there. For more details click on the <em>i<\/em>.","settings_perma":"Activate permanently and approve data transmission:","cookie_path":null,"cookie_expires":"365","css_path":"","services":{"facebook":{"status":"on","dummy_img":".\/css\/images\/dummy_facebook.png","txt_info":"2 clicks for more privacy: only after clicking the button is active and you can sent your recommendation to Facebook. As soon as you activate the button, data will be transmitted to third parties &ndash; see <em>i<\/em>.","txt_facebook_off":"not connected to Facebook","txt_facebook_on":"connected to Facebook","perma_option":"on","display_name":"Facebook","referrer_track":"","language":"en_US","action":"like"},"twitter":{"status":"on","dummy_img":".\/css\/images\/dummy_twitter.png","txt_info":"2 clicks for more privacy: only after clicking the button is active and you can sent your recommendation to Twitter. As soon as you activate the button, data will be transmitted to third parties &ndash; see <em>i<\/em>.","txt_twitter_off":"not connected to Twitter","txt_twitter_on":"connected to Twitter","perma_option":"on","display_name":"Twitter","referrer_track":"","language":"en"},"gplus":{"status":"on","dummy_img":".\/css\/images\/dummy_gplus.png","txt_info":"2 clicks for more privacy: only after clicking the button is active and you can sent your recommendation to Google+. As soon as you activate the button, data will be transmitted to third parties &ndash; see <em>i<\/em>.","txt_gplus_off":"not connected to Google+","txt_gplus_on":"connected to Google+","perma_option":"on","display_name":"Google+","referrer_track":"","language":"en"},"xing":{"status":"on","dummy_img":".\/css\/images\/dummy_xing.png","txt_info":"2 clicks for more privacy: only after clicking the button is active and you can sent your recommendation to XING. As soon as you activate the button, data will be transmitted to third parties &ndash; see <em>i<\/em>.","txt_xing_off":"not connected to XING","txt_xing_on":"connected to XING","perma_option":"on","display_name":"XING","referrer_track":"","language":"en"},"linkedin":{"status":"on","dummy_img":".\/css\/images\/dummy_linkedin.png","txt_info":"2 clicks for more privacy: only after clicking the button is active and you can sent your recommendation to LinkedIn. As soon as you activate the button, data will be transmitted to third parties &ndash; see <em>i<\/em>.","txt_linkedin_off":"not connected to LinkedIn","txt_linkedin_on":"connected to LinkedIn","perma_option":"on","display_name":"LinkedIn","referrer_track":"","language":"en_US"}}});});/* ]]> */</script>